### PR TITLE
Changes to get ficus devstack building through packer

### PR DESCRIPTION
**Pull request by shaidar**
Added the following to the packer build of ficus:
- pressed.cfg to get Ubuntu setup
- some initialization and configurations scripts to bootstrap the instance and get salt as masterless minion operational
- general cleanup

### CODE REVIEW

File: http/ubuntu-16.04/preseed.cfg [hunk 1]
```diff
@@ -0,0 +1,109 @@
+#### Contents of the preconfiguration file (for xenial)
+### Localization
+# Preseeding only locale sets language, country and locale.
+d-i debian-installer/locale string en_US
```
Comment (blarghmatey) [inline http/ubuntu-16.04/preseed.cfg]: Should this be `en_US-UTF8`?

File: http/ubuntu-16.04/preseed.cfg [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+d-i debian-installer/country string US
+
+# netcfg will choose an interface that has link if possible. This makes it
+# skip displaying a list if there is more than one interface.
+d-i netcfg/choose_interface select auto
+
+# If you want to force a hostname, regardless of what either the DHCP
+# server returns or what the reverse DNS entry for the IP is, uncomment
+# and adjust the following line.
+d-i netcfg/hostname string mitodlficus
```
Comment (blarghmatey) [inline http/ubuntu-16.04/preseed.cfg]: Maybe change this to `mitx-devstack`?

File: packer_devstack.json [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      "scripts/common/vagrant.sh",
+      "scripts/common/pkg.sh",
+      "scripts/ubuntu/virtualbox.sh",
+      "scripts/ubuntu/cleanup.sh",
+      "scripts/common/minimize.sh"
+    ]
+  },
+  {
+    "type": "salt-masterless",
+    "local_state_tree": "~/Documents/gitCode/mitodl/salt-ops/salt",
```
Comment (blarghmatey) [inline packer_devstack.json]: This should be a relative path to the packer file. As it is it is dependent on your specific directory structure.

**Review by blarghmatey — CHANGES_REQUESTED** [review]
Overall looks good, just a couple of minor comments.

File: http/ubuntu-16.04/preseed.cfg (hunk 1 above)
Comment (shaidar) [reply http/ubuntu-16.04/preseed.cfg]: Not sure, but I can try it out and change it if that's the case.

Comment (shaidar) [reply http/ubuntu-16.04/preseed.cfg]: That seems to work by changing it to en_US.UTF-8

File: http/ubuntu-16.04/preseed.cfg (hunk 2 above)
Comment (shaidar) [reply http/ubuntu-16.04/preseed.cfg]: changed

File: packer_devstack.json (hunk 3 above)
Comment (shaidar) [reply packer_devstack.json]: changed

File: Vagrantfile [hunk 4]
```diff
@@ -1,12 +1,20 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
+
+$script = <<SCRIPT
+COURSE=#{ENV['course']}
+sudo chown -R www-data /tmp/edx/$COURSE/
+cd /edx/app/edxapp/edx-platform
+sudo -u www-data /edx/bin/python.edxapp ./manage.py cms --settings=aws import /edx/var/edxapp/data /tmp/edx/$COURSE
```
Comment (blarghmatey) [inline Vagrantfile:8]: I think we want to use the import command from lms. It appears that the cms import command is for the contentstore app, whereas the lms import is for courseware...

Comment (shaidar) [reply Vagrantfile:8]: When I tried it using lms, I couldn't get the course to show up in lms or studio even after pre-creating the course in studio. However, by running the above which I found [here](https://openedx.atlassian.net/wiki/spaces/OpenOPS/pages/60227913/Managing+OpenEdX+Tips+and+Tricks), It seems to work as expected (course shows up in both lms and cms) without having to pre-create it.

**Pull request merged**